pkg/maps/lbmap/maglev.go

```diff
@@ -60,6 +68,55 @@ func InitMaglevMaps(ipv4 bool, ipv6 bool) error {
 	return nil
 }
 
+// deleteMapIfMNotMatch removes the outer maglev maps if the M param
+// (MaglevTableSize) has changed. This is to avoid the verifier error when
+// loading BPF programs which access the maps.
+func deleteMapIfMNotMatch(mapName string) (bool, error) {
+	deleteMap := false
+
+	m, err := bpf.OpenMap(mapName)
+	if err == nil {
+		outerKey := &MaglevOuterKey{}
+		if err := bpf.GetNextKey(m.GetFd(), nil, unsafe.Pointer(outerKey)); err == nil {
+			outerVal, err := m.Lookup(outerKey)
+			if err != nil {
+				return false, err
+			}
+			v := outerVal.(*MaglevOuterVal)
+
+			fd, err := bpf.MapFdFromID(int(v.FD))
+			if err != nil {
+				return false, err
+			}
+			info, err := bpf.GetMapInfoByFd(uint32(fd))
+			if err != nil {
+				return false, err
+			}
+			previousM := int(info.ValueSize) / int(unsafe.Sizeof(uint16(0)))
+			if option.Config.MaglevTableSize != previousM {
+				deleteMap = true
+			}
+		} else if err == io.EOF {
+			// The map is empty. To be on the safe side, remove it to avoid M
+			// mismatch. The removal is harmless, as no new entry can be
+			// created while the initialization of the map has not returned.
+			deleteMap = true
+		} else {
+			return false, err
+		}
+	}
+
+	if deleteMap {
+		log.WithField(logfields.BPFMapName, mapName).
+			Info("Deleting Maglev outer map due to possibly different M")
```

Comment:
Why "possibly"? Doesn't the above check guarantee that M changed?

Reply:
Maybe `err == io.EOF` is meant here but agree with Paul

Reply:
The map is deleted in one of the following cases:

1. There was entry in the outer map, and from that entry we can derive the previous M which is != the current M.
2. The outer map was empty, so we could not derive the previous M. Therefore, we don't know whether it was different.

Because of the 2. case I added "possibly" to the log msg.

Reply:
It's really a nit, but: `due to different M or empty map`?

Reply:
Let's wait until the CI tests have finished, then I will push the suggestion.